### PR TITLE
isRepeatData and pinSlot editor implementation

### PR DESCRIPTION
**Pull request by maurofmferrao**
relates to xibosignageltd/xibo-private#371

### CODE REVIEW

File: views/common.twig [hunk 1]
```diff
@@ -503,6 +503,8 @@
                 layer: "{{ "Layer"|trans }}",
                 dataSlot: "{{ "Data Slot"|trans }}",
                 dataSlotHelpText: "{{ "When there are more than one of the same element for a widget you can set the slot for each element. For example with two of the same element you'd have data slot 1 and data slot 2. If 10 items were returned slot 1 would receive items 1,3,5,7,9 and slot 2 would receive items 2,4,6,8,19."|trans }}",
+                pinSlot: "{{ "Pin Slot"|trans }}",
+                pinSlotHelpText: "{{ "Explain this please!!!!."|trans }}",
```
Comment (dasgarner) [inline views/common.twig:507]: We ought to get this set before merging.

Comment (maurofmferrao) [reply views/common.twig:507]: Oh sure, totally forgot about it, thank you Dan!

**Review by rubenpingol-xibo — APPROVED** [review]
Looks good to me!

File: ui/src/templates/forms/widget.hbs [hunk 2]
```diff
@@ -20,6 +20,10 @@
                     {{> inputs/checkbox value=commonProperties.useDuration id=useDuration name="useDuration" title="Set a duration?" helpText="Select to provide a specific duration for this Widget" }}
                     {{> inputs/number value=commonProperties.duration id=duration name="duration" title="Duration" helpText="The duration of the widget in seconds" visibility='{ "field": "useDuration", "type": "eq", "value": "1" }' }}
                     {{> inputs/dropdown value=commonProperties.enableStat options=(arr (obj name='Off' title='Off') (obj name='On' title='On') (obj name='Inherit' title='Inherit')) optionsTitle="title" optionsValue="name" id=enableStat name="enableStat" title="Enable Stats Collection?" helpText="Enable the collection of Proof of Play statistics for this Widget. Ensure that ‘Enable Stats Collection’ is set to ‘On’ in the Display Settings." }}
+
+                    {{#if repeatDataActive}}
+                        {{> inputs/checkbox value=commonProperties.isRepeatData id=isRepeatData name="isRepeatData" title="Is Repeat Data?" helpText="// TODO: Write something here!" }}
```
Comment (dasgarner) [inline ui/src/templates/forms/widget.hbs:25]: Repeat Items?
Once all data items have been placed into slots, any empty slots will be filled with items from the start.

File: views/common.twig (hunk 1 above)
Comment (dasgarner) [reply views/common.twig:507]: Pin Item?
Once a data item has been assigned to this slot it will be pinned and will not cycle with the other data slots.

**Review by dasgarner — APPROVED** [review]

**Pull request merged**